app/src/main/java/com/example/sharewhatyoucanproject/HomescreenActivity.kt

```diff
@@ -5,21 +5,45 @@ import android.os.Bundle
 import android.content.Intent
 import android.view.View
 import android.widget.ImageView
+import com.google.firebase.auth.FirebaseAuth
 
 class HomescreenActivity : AppCompatActivity() {
+    var firebaseAuth: FirebaseAuth? = null
+    var donerimg: ImageView? = null
+    var receiverimg: ImageView? = null
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_homescreen)
-        //donor action
-        val button = findViewById<View>(R.id.donerimg) as ImageView
-        button.setOnClickListener { // Do something in response to button click
-            startActivity(Intent(this@HomescreenActivity, LoginActivity::class.java))
-        }
 
-        //receiver action
```

Comment:
Nitpick, but code comments are easier to read when you leave a space between the `//` and the text.  So, here, of example, should be `// received action` instead of `//received action`.